bot/utils/redis_cache.py

```diff
@@ -41,7 +39,39 @@ def _set_namespace(self, namespace: str) -> None:
         self._namespaces.append(namespace)
         self._namespace = namespace
 
-    def __set_name__(self, owner: object, attribute_name: str) -> None:
+    @staticmethod
+    def _to_typestring(value: ValidRedisType) -> str:
+        """Turn a valid Redis type into a typestring."""
+        if isinstance(value, float):
+            return f"f|{value}"
+        elif isinstance(value, int):
+            return f"i|{value}"
+        elif isinstance(value, str):
+            return f"s|{value}"
+
+    @staticmethod
+    def _from_typestring(value: str) -> ValidRedisType:
+        """Turn a valid Redis type into a typestring."""
+        if value.startswith("f|"):
+            return float(value[2:])
```

Comment:
I'm assuming that empty type strings of float/int type are impossible here? Otherwise this may raise a `ValueError` for trying to `float("")` or `int("")`. I don't know enough about redis to be sure of that.

Reply:
hmm.. I can't imagine what an empty float or int would look like. I don't think that's possible to create, no.

Reply:
I don't know either and I can't think of a situation where we would set such an empty `int`/`float` string with our serialization method. I came across this before I read the `_to_typestring`.

Reply:
I'll resolve this then.